library/summary.ml

```diff
@@ -61,6 +64,18 @@ let empty_frozen = { summaries = Frozen.empty; ml_module = None }
 
 module HMap = Dyn.HMap(Decl)(ID)
 
+let freeze_staged_summaries stage ~marshallable : frozen =
+  let filter = { HMap.filter = fun tag decl -> decl.stage == stage } in
```

Comment:
Don't use pointer equality, it's not robust against changes to the stage type. Instead, roll out your own equality check by pattern-matching.

Reply:
Done.